tests/PHPUnit/Unit/Config/ConfigTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Piwik\Tests\Unit\Config;
+
+use Piwik\Application\Kernel\GlobalSettingsProvider;
+use Piwik\Config;
+use Piwik\Config\ConfigNotFoundException;
+use PHPUnit\Framework\TestCase;
+use Piwik\Container\StaticContainer;
+
+class ConfigTest extends TestCase
+{
+
+    private $config;
+
+    protected function setUp(): void
+    {
+        $userSettingsPath = __DIR__ . '/test_files/boolean_settings_test.ini.php';
+        $settingsProvider = new GlobalSettingsProvider([], $userSettingsPath);
+        $this->config = new Config($settingsProvider);
+    }
+
+    public function testGetBool_true()
+    {
+        $config = $this->config;
+
+        $this->assertTrue($this->getBool('BoolSettings', 'one'));
+        $this->assertTrue($this->getBool('BoolSettings', 'onestr'));
+        $this->assertTrue($this->getBool('BoolSettings', 'truebool'));
+        $this->assertTrue($this->getBool('BoolSettings', 'isyes'));
+        $this->assertTrue($this->getBool('BoolSettings', 'ison'));
+    }
+
+    public function testGetBool_false()
+    {
+        $this->assertFalse($this->getBool('BoolSettings', 'truestr'));
+        $this->assertFalse($this->getBool('BoolSettings', 'two'));
+        $this->assertFalse($this->getBool('BoolSettings', 'twostr'));
+        $this->assertFalse($this->getBool('BoolSettings', 'invalid'));
+        $this->assertFalse($this->getBool('BoolSettings', 'oneinstr'));
+        $this->assertFalse($this->getBool('BoolSettings', 'twoinstr'));
+        $this->assertFalse($this->getBool('BoolSettings', 'isoff'));
+        $this->assertFalse($this->getBool('BoolSettings', 'isno'));
+        $this->assertFalse($this->getBool('BoolSettings', 'truestr'));
+    }
+
+    public function testGetBool_section_Exception()
```

Comment:
fyi @geekdenz because the  getBool is no production method anymore we wouldn't need these edge case tests  and also could simplify the `getBool` method to not throw any exception

Reply:
removed the exception code and edge case